### PR TITLE
CNFT2-1467 Manage Tabs - edit tab in modal

### PR DESCRIPTION
**Pull request by jlee-enquizit**
Description
Here we add the tab editing functionality to the Manage Tabs modal.
Also took this opportunity to do some refactor and add alert banners when adding a new tab as well.
https://www.figma.com/file/Uz9YzHbHQAubHpQTcsibA2/NBS-PageBuilder-2.0?node-id=3526%3A109147&mode=dev

Tickets
https://cdc-nbs.atlassian.net/browse/CNFT2-1467?atlOrigin=eyJpIjoiZGY3NGZlNmU5ODQ5NGNmNzhjMTM2NDZmYWFjZjlmNDkiLCJwIjoiaiJ9

Steps to verify
From Page Builder Pages Library:
1. Select a Page by clicking the Page name
2. In the Tabs menu bar, select `Manage Tabs`
3. Click on the `Edit` (pencil icon) button to open the details in edit mode
4. Create changes to the name or visibility
5. Click save to bring up a confirmation alert, the tab edits should be shown in the Tabs menu bar.

### CODE REVIEW

**Comment by mpeels** [comment]
> Here we add the tab editing functionality to the Manage Tabs modal. Also took this opportunity to do some refactor and add alert banners when adding a new tab as well.

Can you update the PR to follow the pr template?

File: apps/modernization-ui/src/apps/page-builder/pages/EditPage/EditPageTabs/EditPageTabs.tsx [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+                        'Manage Tabs'
+                    )
+                }
+                modalBody={
+                    <div className="edit-page-tabs__modal--body">
+                        {addSuccess ? (
+                            <AlertBanner type="success">Successfully added {tabDetails.name}</AlertBanner>
+                        ) : editSuccess ? (
+                            <AlertBanner type="success" onClose={() => resetEditPageTabs()}>
+                                Successfully edited {selectedEditTab?.name}
```
Comment (mpeels) [inline apps/modernization-ui/src/apps/page-builder/pages/EditPage/EditPageTabs/EditPageTabs.tsx:138]: Issue: Success message does not match Figma. 
Current:
image

Figma:
image


Question: When changing the name of a tab, should the message display the `new` tab name? it is currently displaying the previous name.

File: apps/modernization-ui/src/apps/page-builder/pages/EditPage/EditPageTabs/EditPageTabs.tsx [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+                            </Button>
+                        </div>
+                    ) : (
+                        'Manage Tabs'
+                    )
+                }
+                modalBody={
+                    <div className="edit-page-tabs__modal--body">
+                        {addSuccess ? (
+                            <AlertBanner type="success">Successfully added {tabDetails.name}</AlertBanner>
```
Comment (mpeels) [inline apps/modernization-ui/src/apps/page-builder/pages/EditPage/EditPageTabs/EditPageTabs.tsx:135]: Issue: [Figma](https://www.figma.com/file/Uz9YzHbHQAubHpQTcsibA2/NBS-PageBuilder-2.0?type=design&node-id=6809-240701&mode=dev) shows the message `You’ve successfully Added a new tab!` 


image

While we are showing `Successfully added {tabDetails.name}`. I like this message better but it does not match design.

File: apps/modernization-ui/src/apps/page-builder/pages/EditPage/ManageTabs/ManageTabs.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+const ManageTabs = ({ tabs, setSelectedEditTab }: Props) => {
+    return (
+        <div className="manage-tabs">
+            {tabs.map((tab, i) => {
+                return (
+                    <div key={i} className="manage-tabs__tile" style={{ listStyle: 'none' }}>
+                        <div className="manage-tabs__label">
+                            <IconComponent name="drag" />
+                            <IconComponent name="folder" />
+                            {tab.name}
```
Comment (mpeels) [inline apps/modernization-ui/src/apps/page-builder/pages/EditPage/ManageTabs/ManageTabs.tsx:21]: Issue: Design shows the number of elements within a tab alongside its name
image

**Review by mpeels — CHANGES_REQUESTED** [review]

File: apps/modernization-ui/src/apps/page-builder/pages/EditPage/EditPageTabs/EditPageTabs.tsx [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+                .then(() => {
+                    setAddSuccess(true);
+                    setIsAdding(false);
+                    onAddSuccess();
+                })
+                .catch((e) => {
+                    console.error(e);
+                });
+        }
+    };
```
[the remark this answers is not in the document]
Comment (jlee-enquizit) [reply apps/modernization-ui/src/apps/page-builder/pages/EditPage/EditPageTabs/EditPageTabs.tsx:64]: I can refactor this in the next branch which includes deleting and adding tabs?

File: apps/modernization-ui/src/apps/page-builder/pages/EditPage/ManageTabs/ManageTabs.tsx (hunk 3 above)
Comment (jlee-enquizit) [reply apps/modernization-ui/src/apps/page-builder/pages/EditPage/ManageTabs/ManageTabs.tsx:21]: This is still being debated, like the number tiles in the Edit Page view.
Looking like users prefer the number explained but designs are still pending.

**Pull request closed without merging**